src/raft.c

```diff
@@ -187,9 +187,11 @@ static ShardGroup *GetSlotShardGroup(RedisRaftCtx *rr, int slot, bool asking)
 }
 
 static RRStatus validateRaftRedisCommandArray(RedisRaftCtx *rr, RedisModuleCtx *reply_ctx,
-                                              RaftRedisCommandArray *cmds, unsigned int slot)
+                                              RaftRedisCommandArray *cmds)
 {
-    RedisModule_Assert(slot <= REDIS_RAFT_HASH_MAX_SLOT);
+    RedisModule_Assert(0 <= cmds->slot && cmds->slot <= REDIS_RAFT_HASH_MAX_SLOT);
+
+    unsigned int slot = (unsigned int) cmds->slot;
```

Comment:
`-1` is used as logical "no keys".  Maybe, we can define `NO_KEYS = -1` and use the variable unsigned. Then, check like `if (slot == NO_KEYS)`;

Reply:
actually, we don't use slot value for validation at all, since we return an error from `computeHashSlot` - `-1` is used only to mark that we are handling the first key in `computeHashSlot` - so I guess we can use a local flag for that...